internal/dataplane/parser/translate_httproute.go

```diff
@@ -221,11 +231,26 @@ func generateKongRouteFromTranslation(
 	httproute *gatewayv1beta1.HTTPRoute,
 	translation translators.KongRouteTranslation,
 	addRegexPrefix bool,
+	expressionRoutes bool,
 ) ([]kongstate.Route, error) {
 	// gather the k8s object information and hostnames from the httproute
 	objectInfo := util.FromK8sObject(httproute)
 	tags := util.GenerateTagsForObject(httproute)
 
+	// translate to expression based routes when expressionRoutes is enabled.
+	if expressionRoutes {
+		// get the hostnames from the HTTPRoute
+		hostnames := getHTTPRouteHostnamesAsSliceOfStrings(httproute)
+		return translators.GenerateKongExpressionRoutesFromHTTPRouteMatches(
+			translation.Name,
+			translation.Matches,
+			translation.Filters,
```

Comment:
I think we could just pass a `translation` here instead of unpacking all of its fields.

Reply:
done